strawberryfields/backends/states.py

```diff
@@ -914,6 +914,23 @@ def number_expectation(self, modes):
             ps = np.tensordot(np.diag(values), ps, axes=((0, 1), (0, 1)))
         return float(ps)
 
+    def parity_expectation(self, mode):
+
+        if (type(mode) != int):
+            raise ValueError("Specified mode index must be an integer.")
+
+        cutoff = self._cutoff  # Fock space cutoff.
+        num_modes = self._modes # number of modes in the state.
+        values = np.arange(cutoff)
+
+        ps = self.all_fock_probs()
+        sum_axes = list(range(num_modes))
+        del sum_axes[mode]
+        ps = np.sum(ps, axis=tuple(sum_axes))
+        vals = [(-1)**i for i in values]
+        ev = np.dot(ps, vals)
+        
+        return float(ev)
```

Comment:
This is looking good! @nquesada has some ideas on how to extend and vectorize this :)